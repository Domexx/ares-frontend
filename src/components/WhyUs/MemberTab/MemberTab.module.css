/* === === === === WhyUs/MemberTab === === === === */

.MemberTab {
  margin-top: 48px;
}

.MemberTab-item {
  display: flex;
  flex-direction: column;
  grid-column: 1 / span 14;
  gap: 15px;
  margin-top: 32px;
}

.MemberTab-avatar {
  overflow: hidden;
  width: 72px;
  height: 72px;
  border-radius: 50%;
  background: var(--clr-dark-400);
}

.MemberTab-itemContent {
  display: flex;
  gap: 30px;
  align-items: center;
  padding: 10px;
  border-radius: 12px;
  background: var(--clr-dark-600);
}

.MemberTab-itemDescription {
  color: var(--clr-dark-200);
}

@media (--tablet-landscape) {
  .MemberTab-item {
    grid-column: 1 / span 16;
  }

  .MemberTab-itemDescription {
    font-size: 14px;
  }
}

@media (--tablet) {
  .MemberTab-itemContent {
    flex-direction: column;
    padding: 20px;
  }

  .MemberTab-itemDescription {
    font-size: 14px;
  }
}

@media (--phone) {
  .MemberTab {
    margin-top: 5px;
  }

  .MemberTab-item {
    grid-column: 2 / span 14;
    margin-top: 50px;
  }

  .MemberTab-itemContent {
    flex-direction: column;
    padding: 20px;
  }

  .MemberTab-itemDescription {
    font-size: 14px;
  }
}
