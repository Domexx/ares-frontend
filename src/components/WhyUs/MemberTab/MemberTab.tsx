import React from 'react';
import Image from 'next/image';
import { Member, Team, TeamItem } from '../../../shared/interfaces/WhyUsItem';
import s from './MemberTab.module.css';

export type Props = {
  main: Team;
};

const MemberTab: React.FC<Props> = ({ main }) => {
  const items = main.items.map((item: TeamItem, index: number) => {
    return (
      // eslint-disable-next-line react/no-array-index-key
      <div key={index} className={s['MemberTab-item']}>
        <h1 className="title">{item.title}</h1>
        {item.member.map((member: Member, key: number) => {
          return (
            // eslint-disable-next-line react/no-array-index-key
            <div key={key} className={s['MemberTab-itemContent']}>
              <div className={s['MemberTab-avatar']}>
                <Image src={member.look} layout="fixed" width={64} height={110} alt={member.name} />
              </div>
              <div className={`${s['MemberTab-itemName']} title`}>{member.name}</div>
              <p className={s['MemberTab-itemDescription']}>{member.profession_text}</p>
            </div>
          );
        })}
      </div>
    );
  });
  return <div className={`${s['MemberTab']} u-grid`}>{items}</div>;
};

export default MemberTab;
