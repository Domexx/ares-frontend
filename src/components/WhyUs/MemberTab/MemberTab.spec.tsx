import React from 'react';

import { render } from '@testing-library/react';

import MemberTab from './MemberTab';

describe('Member Tab Component', () => {
  it('should render properly', () => {
    const { container } = render(
      <MemberTab
        main={{
          title: 'Team',
          items: [
            {
              title: 'Council:',
              member: [
                {
                  look: 'https://imager.habboon.pw/?figure=hr-100.hd-180-1.ch-210-66.lg-270-82.sh-290-91&direction=2&head_direction=3&gesture=sml&action=wav',
                  name: 'Virtual',
                  profession_text:
                    'Jelly sweet tart icing muffin wafer. Tootsie roll lemon drops oat cake chupa chups gummies.',
                },
                {
                  look: 'https://imager.habboon.pw/?figure=hr-100.hd-180-1.ch-210-66.lg-270-82.sh-290-91&direction=2&head_direction=3&gesture=sml&action=wav',
                  name: 'Virtual',
                  profession_text:
                    'Jelly sweet tart icing muffin wafer. Tootsie roll lemon drops oat cake chupa chups gummies.',
                },
                {
                  look: 'https://imager.habboon.pw/?figure=hr-100.hd-180-1.ch-210-66.lg-270-82.sh-290-91&direction=2&head_direction=3&gesture=sml&action=wav',
                  name: 'Virtual',
                  profession_text:
                    'Jelly sweet tart icing muffin wafer. Tootsie roll lemon drops oat cake chupa chups gummies.',
                },
              ],
            },
            {
              title: 'Developer:',
              member: [
                {
                  look: 'https://imager.habboon.pw/?figure=hr-100.hd-180-1.ch-210-66.lg-270-82.sh-290-91&direction=2&head_direction=3&gesture=sml&action=wav',
                  name: 'Virtual',
                  profession_text:
                    'Jelly sweet tart icing muffin wafer. Tootsie roll lemon drops oat cake chupa chups gummies.',
                },
                {
                  look: 'https://imager.habboon.pw/?figure=hr-100.hd-180-1.ch-210-66.lg-270-82.sh-290-91&direction=2&head_direction=3&gesture=sml&action=wav',
                  name: 'Virtual',
                  profession_text:
                    'Jelly sweet tart icing muffin wafer. Tootsie roll lemon drops oat cake chupa chups gummies.',
                },
                {
                  look: 'https://imager.habboon.pw/?figure=hr-100.hd-180-1.ch-210-66.lg-270-82.sh-290-91&direction=2&head_direction=3&gesture=sml&action=wav',
                  name: 'Virtual',
                  profession_text:
                    'Jelly sweet tart icing muffin wafer. Tootsie roll lemon drops oat cake chupa chups gummies.',
                },
              ],
            },
          ],
        }}
      />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
