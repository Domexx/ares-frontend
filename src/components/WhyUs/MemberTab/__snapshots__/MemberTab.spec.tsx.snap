// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member Tab Component should render properly 1`] = `
<div
  class="MemberTab u-grid"
>
  <div
    class="MemberTab-item"
  >
    <h1
      class="title"
    >
      Council:
    </h1>
    <div
      class="MemberTab-itemContent"
    >
      <div
        class="MemberTab-avatar"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 64px; height: 110px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <img
            alt="Virtual"
            data-nimg="fixed"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <div
        class="MemberTab-itemName title"
      >
        Virtual
      </div>
      <p
        class="MemberTab-itemDescription"
      >
        Jelly sweet tart icing muffin wafer. Tootsie roll lemon drops oat cake chupa chups gummies.
      </p>
    </div>
    <div
      class="MemberTab-itemContent"
    >
      <div
        class="MemberTab-avatar"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 64px; height: 110px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <img
            alt="Virtual"
            data-nimg="fixed"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <div
        class="MemberTab-itemName title"
      >
        Virtual
      </div>
      <p
        class="MemberTab-itemDescription"
      >
        Jelly sweet tart icing muffin wafer. Tootsie roll lemon drops oat cake chupa chups gummies.
      </p>
    </div>
    <div
      class="MemberTab-itemContent"
    >
      <div
        class="MemberTab-avatar"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 64px; height: 110px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <img
            alt="Virtual"
            data-nimg="fixed"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <div
        class="MemberTab-itemName title"
      >
        Virtual
      </div>
      <p
        class="MemberTab-itemDescription"
      >
        Jelly sweet tart icing muffin wafer. Tootsie roll lemon drops oat cake chupa chups gummies.
      </p>
    </div>
  </div>
  <div
    class="MemberTab-item"
  >
    <h1
      class="title"
    >
      Developer:
    </h1>
    <div
      class="MemberTab-itemContent"
    >
      <div
        class="MemberTab-avatar"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 64px; height: 110px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <img
            alt="Virtual"
            data-nimg="fixed"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <div
        class="MemberTab-itemName title"
      >
        Virtual
      </div>
      <p
        class="MemberTab-itemDescription"
      >
        Jelly sweet tart icing muffin wafer. Tootsie roll lemon drops oat cake chupa chups gummies.
      </p>
    </div>
    <div
      class="MemberTab-itemContent"
    >
      <div
        class="MemberTab-avatar"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 64px; height: 110px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <img
            alt="Virtual"
            data-nimg="fixed"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <div
        class="MemberTab-itemName title"
      >
        Virtual
      </div>
      <p
        class="MemberTab-itemDescription"
      >
        Jelly sweet tart icing muffin wafer. Tootsie roll lemon drops oat cake chupa chups gummies.
      </p>
    </div>
    <div
      class="MemberTab-itemContent"
    >
      <div
        class="MemberTab-avatar"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 64px; height: 110px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <img
            alt="Virtual"
            data-nimg="fixed"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <div
        class="MemberTab-itemName title"
      >
        Virtual
      </div>
      <p
        class="MemberTab-itemDescription"
      >
        Jelly sweet tart icing muffin wafer. Tootsie roll lemon drops oat cake chupa chups gummies.
      </p>
    </div>
  </div>
</div>
`;
